forte/data/selector.py

```diff
@@ -66,12 +68,32 @@ def select(self, pack: MultiPack) -> Iterator[DataPack]:
 class NameMatchSelector(SinglePackSelector):
     r"""Select a :class:`DataPack` from a :class:`MultiPack` with specified
     name.
+
+    This implementation takes special care for backward compatability:
+    Deprecated:
+        selector = NameMatchSelector(select_name="foo")
+        selector = NameMatchSelector("foo")
+    Now:
+        selector = NameMatchSelector(
+            configs={
+                "select_name": "foo"
+            }
+        )
     """
 
-    def __init__(self, select_name: str):
-        super().__init__()
-        assert select_name is not None
-        self.select_name: str = select_name
+    def __init__(self, *args, **kwargs):
```

Comment:
To preserve backward compatibility is easy here, we simply need to load `self._select_name` from the args, and then in the `initialize` function, we overwrite `self._select_name`.